sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/AadClientRegistrationRepository.java

```diff
@@ -132,19 +149,28 @@ private Set<String> delegatedClientsAccessTokenScopes(AadAuthenticationPropertie
     }
 
     private ClientRegistration toClientRegistration(String registrationId,
-                                                    AadAuthorizationGrantType aadAuthorizationGrantType,
+                                                    AuthorizationGrantType authorizationGrantType,
                                                     Collection<String> scopes,
+                                                    ClientAuthenticationMethod clientAuthenticationMethod,
                                                     AadAuthenticationProperties properties) {
         AadAuthorizationServerEndpoints endpoints =
-            new AadAuthorizationServerEndpoints(properties.getProfile().getEnvironment().getActiveDirectoryEndpoint(), properties.getProfile().getTenantId());
+            new AadAuthorizationServerEndpoints(properties.getProfile().getEnvironment().getActiveDirectoryEndpoint(),
+                properties.getProfile().getTenantId());
+
+        if (ON_BEHALF_OF.equals(authorizationGrantType)) {
+            authorizationGrantType = JWT_BEARER;
+            LOGGER.warn("The grant type 'on_behalf_of' of the client '{}' is not recommended, it will be "
```

Comment:
why do we warn it here?

Reply:
It means the type value is changed by our logic, I will improve the log content.